### PR TITLE
[Enhancement](maxCompute)Implement compatibility with existing maxcompute catalogs from previous versions.

### PR DESCRIPTION
**Pull request by hubgeter**
before pr #40225
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by hubgeter** [comment]
run buildall

File: fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java [hunk 1]
```diff
@@ -117,6 +118,14 @@ public MaxComputeJniScanner(int batchSize, Map<String, String> params) {
         project = Objects.requireNonNull(params.get(PROJECT), "required property '" + PROJECT + "'.");
         table = Objects.requireNonNull(params.get(TABLE), "required property '" + TABLE + "'.");
         sessionId = Objects.requireNonNull(params.get(SESSION_ID), "required property '" + SESSION_ID + "'.");
+        String timeZoneName = Objects.requireNonNull(params.get(TIME_ZONE), "required property '" + TIME_ZONE + "'.");
+        try {
+            timeZone = ZoneId.of(timeZoneName);
+        } catch (Exception e) {
+            LOG.info(e.getMessage());
+            LOG.info("set timeZoneName = " + timeZoneName + "fail, use systemDefault.");
```
Comment (morningman) [inline fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:126]: use LOG.warn, and use one line.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java [hunk 2]
```diff
@@ -43,13 +48,16 @@
 import java.util.stream.Collectors;
 
 public class MaxComputeExternalCatalog extends ExternalCatalog {
+    private static final String endpointTemplate = "http://service.{}.maxcompute.aliyun-inc.com/api";
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java:51]: Add aliyun doc link about this endpoint.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java [hunk 3]
```diff
@@ -67,11 +75,46 @@ public MaxComputeExternalCatalog(long catalogId, String name, String resource, M
...
     }
 
+    //Compatible with existing catalogs in previous versions.
+    protected void generatorEndpoint() {
+        Map<String, String> props = catalogProperty.getProperties();
+
+        if (props.containsKey(MCProperties.ENDPOINT)) {
+            endpoint = props.get(MCProperties.ENDPOINT);
+            return;
+        } else if (props.containsKey(MCProperties.TUNNEL_SDK_ENDPOINT)) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java:85]: Add some example in comment to explain this conversion.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java [hunk 4]
```diff
@@ -363,11 +363,12 @@ private String convertLiteralToOdpsValues(OdpsType odpsType, Expr expr) throws A
                 ScalarType dstType = ScalarType.createDateV2Type();
                 return  " \"" + dateLiteral.getStringValue(dstType) + "\" ";
             }
-            case DATETIME: {
-                DateLiteral dateLiteral = (DateLiteral) literalExpr;
-                ScalarType dstType = ScalarType.createDatetimeV2Type(3);
-                return  " \"" + dateLiteral.getStringValue(dstType) + "\" ";
-            }
+            // case DATETIME: {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java:366]: why comment out?

File: fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java (hunk 1 above)
Comment (hubgeter) [reply fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:126]: i wiil fix

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java (hunk 2 above)
Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java:51]: ok . i will add it

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java (hunk 3 above)
Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java:85]: ok . i will add it

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java (hunk 4 above)
Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/source/MaxComputeScanNode.java:366]: I should determine which time zone the datetime needs to be passed in.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java [hunk 5]
```diff
@@ -40,14 +40,19 @@
 import com.google.common.collect.ImmutableList;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParser;
+import jdk.internal.org.jline.utils.Log;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java:43]: wrong log import

Comment (hubgeter) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/maxcompute/MaxComputeExternalCatalog.java]: i fix it

**Comment by hubgeter** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**